app/routers/accounts.py

```diff
@@ -1,31 +1,135 @@
-from fastapi import APIRouter, status
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, HTTPException, status
 from fastapi_pagination.limit_offset import LimitOffsetPage
 
+from app.auth.auth import check_operations_account
+from app.db.handlers import NotFoundError
+from app.db.models import Account
+from app.dependencies import AccountId, AccountRepository
+from app.enums import AccountStatus, AccountType
+from app.pagination import paginate
 from app.schemas import (
     AccountCreate,
     AccountRead,
     AccountUpdate,
     AccountUserRead,
+    from_orm,
+    to_orm,
 )
 
 router = APIRouter()
 
 
-@router.post("", response_model=AccountRead, status_code=status.HTTP_201_CREATED)
-async def create_account(
-    data: AccountCreate,
-):
-    pass
+async def persist_data_and_format_response(account_repo, data):
+    """
+    It persists the given data to the Account model and return back
+    the Pydantic schema
 
+    account_repo: an ORM model instance of AccountRepository
+    data: the data to persist
+    Return: AccountRead Pydantic Model
+    """
+    account = to_orm(data, Account)
+    db_account = await account_repo.create(account)
+    return from_orm(AccountRead, db_account)
 
-@router.get("", response_model=LimitOffsetPage[AccountRead])
-async def get_accounts():
-    pass
+
+async def validate_account_type_and_required_conditions(account_repo, data):
+    """
+    This function performs the following required checks before
+    proceeding to create an Account:
+    1. Only Accounts classified as of type “Affiliate” can be created.
+    2. The external_id field has to be unique and not in DELETED status.
+
+    A HTTPException with status 400 will be raised if at least one condition is not met.
+    """
+    if data.type != AccountType.AFFILIATE:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="An Account with external ID " f"`{data.external_id}` already exists.",
```

Comment:
This error message is wrong. It should say
"You cannot create an Account of type Operations" or "Only Affiliate type accounts can be created".